JetUtilities/src/crystalBall.cc

```diff
@@ -0,0 +1,227 @@
+
+  // remove discontinuity at x=R1: solve pol7=exp for p0 left 
+  double p0_left = exp_fcn(R1, normExp_left, slopeExp_left) -
+    ( p1_left*R1 + p2_left*pow(R1,2) +
+      p3_left*pow(R1,3) + p4_left*pow(R1,4) +
+      p5_left*pow(R1,5) + p6_left*pow(R1,6) +
+      p7_left*pow(R1,7) ) ;
+
+  double result = 0.;
+  if (drawPolLeft  && x >= 0.        && x < R1       ) result += pol_fcn(x, p0_left, p1_left, p2_left, p3_left, p4_left, p5_left, p6_left, p7_left);
+  if (drawExp      && x >= R1        && x < R2       ) result += exp_fcn(x, normExp_left, slopeExp_left);
+  if (drawGaus1    && x >= R2        && x < meanGaus1) result += gauss_fcn(x, normGaus1, meanGaus1, sigGaus1_left);
+  if (drawAnyRight && x >= meanGaus1 && x < 2.       ) result += sum_gaus_pol_fnc(xx, pp);
+  return result;
+} // crystalBall
+
+
+double crystalBall_1(double *xx, double *pp)
+{
+  double x = xx[0];
+
+  // left
+  double normGaus1     = pp[0];  
+  double meanGaus1     = pp[1];  
+  double sigGaus1_left = pp[2];   
+
+  double slopeExp_left = pp[4];  
+
+  double p1_left       = pp[6];
+
+  double p1_right      = pp[17]; 
+
+  double R2            = meanGaus1 - 0.5*sigGaus1_left - pp[8]; 
+  double R1            = R2 - pp[7]; // min exp
+  double R3            = pp[18];
+
+  bool drawPolLeft     = pp[19];
+  bool drawExp         = pp[20];
+  bool drawGaus1       = pp[21];
+  bool drawGaus2       = pp[22];
+  bool drawGaus3       = pp[23];
+  bool drawPolRight    = pp[24];
+  bool drawAnyRight    = drawGaus1 || drawGaus2 || drawGaus3;
+
+  // remove discontinuity at x=R2: solve exp=gauss for norm exp and replace R2 by (meanGaus1-R2*R2)
+  double normExp_left = gauss_fcn(R2, normGaus1, meanGaus1, sigGaus1_left)/TMath::Exp(slopeExp_left*R2);
+
+  // remove discontinuity at x=R1: solve pol1=exp for p0 left 
+  double p0_left = exp_fcn(R1, normExp_left, slopeExp_left) - p1_left*R1 ;
+
+  // remove discontinuity at x=R3: solve pol1=sumGaus for p0 right
+  double r3[]{R3};
+  double p0_right = sum_gaus_fnc(r3, pp)-p1_right*R3;
+
+  // compute probability
+  double result = 0.;
+  if (drawPolLeft  && x >= 0.        && x < R1       ) result += pol_fcn  (x, p0_left, p1_left);
+  if (drawExp      && x >= R1        && x < R2       ) result += exp_fcn  (x, normExp_left, slopeExp_left);
+  if (drawGaus1    && x >= R2        && x < meanGaus1) result += gauss_fcn(x, normGaus1, meanGaus1, sigGaus1_left);
+  if (drawAnyRight && x >= meanGaus1 && x < R3       ) result += sum_gaus_fnc(xx, pp); 
+  if (drawPolRight && x >= R3        && x < 2.       ) result += pol_fcn  (x, p0_right, p1_right);
+
+  return result;
+} // crystalBall_1
+
+
+
+//////////////////////
+double exp_fcn(double x, double norm, double slope){
+  return norm*TMath::Exp(slope*x);
+}
+
+double gauss_fcn(double x, double norm, double mean, double sigma){
+  return norm*TMath::Exp(-0.5*TMath::Power((x - mean)/sigma, 2.));
```

Comment:
You know that TMath has a "TMath::Gaus" function, right? All you need is 
`return norm*TMath::Gaus(x,mean,sigma);`